samplePDF/samplePDFFDBase.cpp

```diff
@@ -340,328 +400,289 @@ void samplePDFFDBase::fillArray() {
       normweight *= CalcXsecWeightNorm(iSample, iEvent);
-
+      
       //DB Catch negative norm weights and skip any event with a negative event. Previously we would set weight to zere and continue but that is inefficient
       if (normweight <= 0.){
-		MCSamples[iSample].xsec_w[iEvent] = 0.;
-	   	continue;
-	  }
-
+        MCSamples[iSample].xsec_w[iEvent] = 0.;
+        continue;
+      }
+      
       funcweight = CalcXsecWeightFunc(iSample,iEvent);
       //DB Catch negative func weights and skip any event with a negative event. Previously we would set weight to zere and continue but that is inefficient
       if (funcweight <= 0.){          
-		MCSamples[iSample].xsec_w[iEvent] = 0.;
-	   	continue;
-	  }
-
-      MCSamples[iSample].xsec_w[iEvent] = splineweight*normweight*funcweight;
-      
-      //DB Set oscillation weights for NC events to 1.0
-      //DB Another speedup - Why bother storing NC signal events and calculating the oscillation weights when we just throw them out anyway? Therefore they are skipped in setupMC
-	  //
-	  //LW Checking if NC event is signal (oscillated or not), if yes: osc_w = 0 || if no: osc_w = 1.0
-      if (MCSamples[iSample].isNC[iEvent] && MCSamples[iSample].signal) { //DB Abstract check on MaCh3Modes to determine which apply to neutral current
-	  	MCSamples[iSample].osc_w[iEvent] = 0.0;
-	  	continue;
+        MCSamples[iSample].xsec_w[iEvent] = 0.;
+        continue;
       }
-
-	  //ETA - I need to check that this doesn't cause a problem for atmospherics and or tau samples
-      if (MCSamples[iSample].isNC[iEvent] && !MCSamples[iSample].signal) { //DB Abstract check on MaCh3Modes to determine which apply to neutral current
-	  	MCSamples[iSample].osc_w[iEvent] = 1.0;
-      }
-	  //DB Set oscillation weights for NC events to 1.0
-	  //DB Another speedup - Why bother storing NC signal events and calculating the oscillation weights when we just throw them out anyway? Therefore they are skipped in setupSKMC
-	  if (MCSamples[iSample].isNC[iEvent]) { //DB Abstract check on MaCh3Modes to determine which apply to neutral current
-		MCSamples[iSample].osc_w[iEvent] = 1.0;	
-	  }
+      
+      MCSamples[iSample].xsec_w[iEvent] = splineweight*normweight*funcweight;
       
       //DB Total weight
-	  totalweight = GetEventWeight(iSample,iEvent);
+      totalweight = GetEventWeight(iSample,iEvent);
       //DB Catch negative weights and skip any event with a negative event
       if (totalweight <= 0.){
-		MCSamples[iSample].xsec_w[iEvent] = 0.;
-	   	continue;
-	  }
- 
+        MCSamples[iSample].xsec_w[iEvent] = 0.;
+        continue;
+      }
       //DB Switch on BinningOpt to allow different binning options to be implemented
       //The alternative would be to have inheritance based on BinningOpt
       double XVar = *(MCSamples[iSample].x_var[iEvent]);
-
+      
       //DB Find the relevant bin in the PDF for each event
       int XBinToFill = -1;
```

Comment:
seems that shift systematics do only work in X, is that clearly documented somewhere?